tests/unit/test_solvers/test_idaklu_solver.py

```diff
@@ -443,6 +447,57 @@ def test_dae_solver_algebraic_model(self):
             solution = solver.solve(model, t_eval)
             np.testing.assert_array_equal(solution.y, -1)
 
+    def test_options(self):
+        model = pybamm.BaseModel()
+        u = pybamm.Variable("u")
+        v = pybamm.Variable("v")
+        model.rhs = {u: -0.1 * u}
+        model.algebraic = {v: v - u}
+        model.initial_conditions = {u: 1, v: 1}
+        disc = pybamm.Discretisation()
+        disc.process_model(model)
+
+        t_eval = np.linspace(0, 1)
+        solver = pybamm.IDAKLUSolver()
+        soln_base = solver.solve(model, t_eval)
+
+        # test print_stats
+        solver = pybamm.IDAKLUSolver(options={"print_stats": True})
+        f = io.StringIO()
+        with redirect_stdout(f):
+            solver.solve(model, t_eval)
+        s = f.getvalue()
+        self.assertIn("Solver Stats", s)
+
+        solver = pybamm.IDAKLUSolver(options={"print_stats": False})
+        f = io.StringIO()
+        with redirect_stdout(f):
+            solver.solve(model, t_eval)
+        s = f.getvalue()
+        self.assertEqual(len(s), 0)
+
+        # test everything else
```

Comment:
should "garbage" throw an error?

Reply:
perhaps...? I went with the option of just using a sensible default if a bad option was passed in, while printing a warning (e.g. if you choose jacobian=sparse and linear_solver=dense it just changes the linear_solver to KLU and prints a warning). Happy to reconsider tho....

Reply:
I prefer erroring if inputs aren't compatible. In most cases it will probably be a typo, which is better caught by an error than a warning (e.g. if they have a low level logger on the warning might get lost). If a user has some weird workflow that means they have to pass in incompatible inputs, they can write their own input preprocessing functions